terraform/modules/cluster/addons.tf

```diff
@@ -1,35 +1,17 @@
-data "aws_eks_addon_version" "latest" {
-  for_each = toset(["vpc-cni"])
-
-  addon_name         = each.value
-  kubernetes_version = var.cluster_version
-  most_recent        = true
-}
-
-resource "aws_eks_addon" "vpc_cni" {
-  cluster_name         = module.eks_blueprints.eks_cluster_id
-  addon_name           = "vpc-cni"
-  addon_version        = data.aws_eks_addon_version.latest["vpc-cni"].version
-  resolve_conflicts    = "OVERWRITE"
-  configuration_values = "{\"env\":{\"ENABLE_PREFIX_DELEGATION\":\"true\", \"ENABLE_POD_ENI\":\"true\"}}"
-
-  depends_on = [
-    null_resource.kubectl_set_env
-  ]
-}
-
 locals {
   ebs_csi_blocker = try(module.eks_blueprints_kubernetes_addons.aws_ebs_csi_driver.release_metadata.metadata.status, "")
 }
 
+resource "aws_iam_instance_profile" "karpenter" {
```

Comment:
once https://github.com/aws-ia/terraform-aws-eks-blueprints-addons is ready, this will go away. This [sub-module](https://github.com/terraform-aws-modules/terraform-aws-eks/tree/master/modules/karpenter) provisions all of the necessary AWS infrastructure resources, including this profile. The new addons module will use this plus the helm release for karpenter which will alleviate the need for this resource